File: appnes/src/main/java/nostalgia/appnes/NesApplication.java
package nostalgia.appnes;

import com.arialyy.aria.core.Aria;
import com.liulishuo.filedownloader.FileDownloader;

import nostalgia.framework.BaseApplication;
import nostalgia.framework.base.EmulatorHolder;

public class NesApplication extends BaseApplication {

    @Override
    public void onCreate() {
        super.onCreate();
        EmulatorHolder.setEmulatorClass(NesEmulator.class);
        FileDownloader.setupOnApplicationOnCreate(this);
    }

    @Override
    public boolean hasGameMenu() {
        return true;
    }
}

File: gui/xml/src/main/java/com/woohyman/xml/ui/NesApplication.kt
package com.woohyman.xml.ui

import com.liulishuo.filedownloader.FileDownloader
import com.woohyman.xml.BaseApplication
import dagger.hilt.android.HiltAndroidApp

@HiltAndroidApp
class NesApplication : BaseApplication() {

    override fun onCreate() {
        super.onCreate()
        FileDownloader.setupOnApplicationOnCreate(this)
    }

    override fun hasGameMenu(): Boolean {
        return true
    }
}